tuf_conformance/test_updater_delegation_graphs.py

```diff
@@ -0,0 +1,151 @@
+from dataclasses import astuple, dataclass, field
+from typing import Any
+
+import pytest
+from tuf.api.metadata import (
+    SPECIFICATION_VERSION,
+    TOP_LEVEL_ROLE_NAMES,
+    DelegatedRole,
+    TargetFile,
+    Targets,
+)
+
+from tuf_conformance.client_runner import ClientRunner
+from tuf_conformance.repository_simulator import RepositorySimulator
+from tuf_conformance.simulator_server import SimulatorServer
+
+# DataSet is only here so type hints can be used.
+DataSet = dict[str, Any]
+
+
+@dataclass
+class DelegationTester:
+    delegator: str
+    rolename: str
+    keyids: list[str] = field(default_factory=list)
+    threshold: int = 1
+    terminating: bool = False
+    paths: list[str] | None = field(default_factory=lambda: ["*"])
+    path_hash_prefixes: list[str] | None = None
+
+
+@dataclass
+class TestTarget:
+    rolename: str
+    content: bytes
+    targetpath: str
+
+
+@dataclass
+class TargetTestCase:
+    targetpath: str
+    found: bool
+    visited_order: list[str] = field(default_factory=list)
+
+
+@dataclass
+class DelegationsTestCase:
+    """A delegations graph as lists of delegations and target files
+    and the expected order of traversal as a list of role names."""
+
+    delegations: list[DelegationTester]
+    target_files: list[TestTarget] = field(default_factory=list)
+    visited_order: list[str] = field(default_factory=list)
+
+
+def init_repo(repo: RepositorySimulator, test_case: DelegationsTestCase) -> None:
+    spec_version = ".".join(SPECIFICATION_VERSION)
+    for d in test_case.delegations:
+        if d.rolename in repo.mds:
+            targets = repo.mds[d.rolename].signed
+        else:
+            targets = Targets(1, spec_version, repo.safe_expiry, {}, None)
+        # unpack 'd' but skip "delegator"
+        role = DelegatedRole(*astuple(d)[1:])
+        repo.add_delegation(d.delegator, role, targets)
+
+    for target in test_case.target_files:
+        repo.add_artifact(*astuple(target))
+
+    if test_case.target_files:
+        repo.targets.version += 1
```

Comment:
I think this is unnecessary for reasons mentioned in the other PR

Reply:
Removed in previous PR. Still not required here.